cronitor/celery.py

```diff
@@ -0,0 +1,147 @@
+            elif isinstance(item, solar):
+                # We don't support solar schedules
+                logger.warning("The cronitor-python celery module does not support "
+                               "tasks using solar schedules. Task schedule '{}' will "
+                               "not be monitored".format(name))
+                continue
+            else:
+                logger.warning("The cronitor-python celery module does not support "
+                               "schedules of type `{}`".format(type(item)))
+                continue
+
+            monitors.append({
+                'type': 'job',
+                'key': name,
+                'schedule': cronitor_schedule,
+            })
+
+            headers.update({
+                'x-cronitor-task-origin': 'celerybeat',
+                'x-cronitor-celerybeat-name': name,
+            })
+
+            app.add_periodic_task(entry.schedule,
+                                  # Setting headers in the signature
+                                  # works better then in periodic task options
+                                  app.tasks.get(entry.task).s().set(headers=headers),
+                                  args=entry.args, kwargs=entry.kwargs,
+                                  name=entry.name, **(entry.options or {}))
+
+        # To avoid recursion, since restarting celerybeat will result in this
+        # signal being called again, we disconnect the signal.
+        beat_init.disconnect(celerybeat_startup, dispatch_uid=1)
+
+        # We need to stop and restart celerybeat to get the task updates in place.
+        # This isn't ideal, but seems to work.
+        sender.stop()
+        app.Beat().run()
+        logger.debug("Creating Cronitor monitors: %s", [m['key'] for m in monitors])
+        Monitor.put(monitors)
+
+    beat_init.connect(celerybeat_startup, dispatch_uid=1)
+
+    @task_prerun.connect
+    def ping_monitor_before_task(sender, **kwargs):  # type: (celery.Task, Dict) -> None
+        headers = get_headers_from_task(sender)
+        if 'x-cronitor-celerybeat-name' in headers:
+            monitor = Monitor(headers['x-cronitor-celerybeat-name'])
+            monitor.ping(state=State.RUN, series=sender.request.id)
+
+    @task_success.connect
+    def ping_monitor_on_success(sender, **kwargs):  # type: (celery.Task, Dict) -> None
+        headers = get_headers_from_task(sender)
+        if 'x-cronitor-celerybeat-name' in headers:
+            monitor = Monitor(headers['x-cronitor-celerybeat-name'])
+            monitor.ping(state=State.COMPLETE, series=sender.request.id)
+
+    @task_failure.connect
+    def ping_monitor_on_failure(sender, **kwargs):  # type: (celery.Task, Dict) -> None
+        headers = get_headers_from_task(sender)
+        if 'x-cronitor-celerybeat-name' in headers:
+            monitor = Monitor(headers['x-cronitor-celerybeat-name'])
+            monitor.ping(state=State.FAIL, series=sender.request.id)
```

Comment:
Can we get the reason in the same way we do on `ping_monitor_on_retry`?

Reply:
Good call, I added this